lib/internal/fs/copy/copy.js

```diff
@@ -0,0 +1,468 @@
+'use strict';
+
+// This file is a modified version of the fs-extra's copy method
+// modified in the following ways:
+//
+// - Use of the graceful-fs has been replaced with fs.
+// - Helpers from stat.js have been inlined.
+// - Formatting and style changes to match Node.js' linting rules.
+// - Parameter validation has been added.
+
+const console = require('console');
+const { codes } = require('internal/errors');
+const {
+  ERR_FS_COPY_DIR_TO_NON_DIR,
+  ERR_FS_COPY_EEXIST,
+  ERR_FS_COPY_FIFO_PIPE,
+  ERR_FS_COPY_NON_DIR_TO_DIR,
+  ERR_FS_COPY_SOCKET,
+  ERR_FS_COPY_SYMLINK_TO_SUBDIRECTORY,
+  ERR_FS_COPY_TO_SUBDIRECTORY,
+  ERR_FS_COPY_UNKNOWN,
+} = codes;
+const {
+  os: {
+    errno: {
+      EEXIST,
+      EISDIR,
+      EINVAL,
+      ENOTDIR,
+    }
+  }
+} = internalBinding('constants');
+const fs = require('fs');
+const {
+  chmod,
+  close,
+  copyFile,
+  futimes,
+  lstat,
+  mkdir,
+  open,
+  readdir,
+  readlink,
+  stat,
+  symlink,
+  unlink,
+} = fs;
+const path = require('path');
+const {
+  dirname,
+  join,
+  parse,
+  resolve,
+  sep,
+} = path;
+const { Promise, PromiseAll, PromiseResolve } = primordials;
+const util = require('util');
+const {
+  callbackify,
+  promisify,
+} = util;
+const statAsync = promisify(stat);
+const lstatAsync = promisify(lstat);
+
+function copyFn(src, dest, opts, cb) {
+  // Warn about using preserveTimestamps on 32-bit node
+  if (opts.preserveTimestamps && process.arch === 'ia32') {
+    console.warn('Using the preserveTimestamps option in 32-bit' +
+      'node is not recommended; see' +
+      'https://github.com/jprichardson/node-fs-extra/issues/269'
+    );
```

Comment:
I think this should be `process.emitWarning` instead.

Reply:
Also, should probably link somewhere other than `fs-extra` issue tracker.

Reply:
switched to `process.emitWarning` stopped linking to an issue.